src/sdk/bucketspace_multi_fs.js

```diff
@@ -0,0 +1,399 @@
+
+//TODO:  dup from namespace_fs - need to handle and not dup code
+function isDirectory(ent) {
+    if (!ent) throw new Error('isDirectory: ent is empty');
+    if (ent.mode) {
+        // eslint-disable-next-line no-bitwise
+        return (((ent.mode) & nb_native().fs.S_IFMT) === nb_native().fs.S_IFDIR);
+    } else if (ent.type) {
+        return ent.type === nb_native().fs.DT_DIR;
+    } else {
+        throw new Error(`isDirectory: ent ${ent} is not supported`);
+    }
+}
+
+
+class BucketSpaceMultiFS extends BucketSpaceFS {
+    constructor({fs_root, iam_dir, multi_fs_config_dir}) {
+        super({fs_root});
+        this.fs_root = fs_root;
+        this.iam_dir = iam_dir;
+        this.multi_fs_config_dir = multi_fs_config_dir;
+        this.fs_context = {
+            uid: process.getuid(),
+            gid: process.getgid(),
+            warn_threshold_ms: config.NSFS_WARN_THRESHOLD_MS,
+        };
+    }
+
+    async read_account_by_access_key({ access_key }) {
+        try {
+            console.log('GGG read_account_by_access_key', this.iam_dir, access_key);
+            if (!access_key) throw new Error('no access key');
+            const iam_path = path.join(this.iam_dir, access_key);
+            console.log('GGG read_account_by_access_key', access_key, iam_path);
+            const { data } = await nb_native().fs.readFile(this.fs_context, iam_path);
+            const account = JSON.parse(data.toString());
+            account.name = new SensitiveString(account.name);
+            account.email = new SensitiveString(account.email);
+            for (const k of account.access_keys) {
+                k.access_key = new SensitiveString(k.access_key);
+                k.secret_key = new SensitiveString(k.secret_key);
+            }
+            console.log('GGG read_account_by_access_key', access_key, account);
+            return account;
+        } catch (err) {
+            throw new RpcError('NO_SUCH_ACCOUNT', `Account with access_key not foundSSSSS`, err);
+        }
+    }
+
+    async read_bucket_sdk_info({ name }) {
+        const bucket_path = path.join(this.fs_root, name);
+        const { data } = await nb_native().fs.readFile(this.fs_context, bucket_path + ".json");
+        const bucket = JSON.parse(data.toString());
+        const nsr = {
+            resource: {
+                fs_root_path: "/",
+            },
+            path: bucket.path
+        };
+        bucket.namespace = {
+            read_resources: [nsr],
+            write_resource: nsr,
+        };
+
+        bucket.system_owner = new SensitiveString(bucket.system_owner);
+        bucket.bucket_owner = new SensitiveString(bucket.bucket_owner);
+        return bucket;
+    }
+
+
+    ////////////
+    // BUCKET //
+    ////////////
+
+    async list_buckets(object_sdk) {
+        try {
+            const entries = await nb_native().fs.readdir(this.fs_context, this.fs_root);
```

Comment:
fix to list the config dir / bucket / jsons.
TODO : we need to add pagination support to list buckets for more than 1000 buckets.